### PR TITLE
Update metadata

### PR DESCRIPTION
**Pull request by danenbm**
### Notes
* Getting update-metadata ready for deployment to devnet, not using main branch as we wait for indexer code to be deployed.
* This PR merges the changes from @samwise2 's PR: https://github.com/metaplex-foundation/mpl-bubblegum/pull/3  into a staging branch, splitting the new update instructions into separate files as is now done on main.

### Verification
Some extra verification since the merging of `main` back to `update_metadata` resulted in some changes:
* Verified that JS-solita changes were same as what was in https://github.com/metaplex-foundation/mpl-bubblegum/pull/3 except for expected small changes such as an additional error code and `InstructionName` and `SetDecompressableState` now being removed from IDL.
* Verified that js-solita `main.test.ts` matches what is in https://github.com/metaplex-foundation/mpl-bubblegum/pull/3 except it adds the call for `setDecompressibleStateIx`.
* Umi generation is similar but some Umi updates result in different code generation (see detailed comments on files).
* Rust client changes are new since merging to main since https://github.com/metaplex-foundation/mpl-bubblegum/pull/3 had no Rust client.
* Bubblegum program error.rs and `UpdateArgs` type is exact same as https://github.com/metaplex-foundation/mpl-bubblegum/pull/3
* Validated `update_metadata` and `update_metadata_collection_nft` and all the helper functions all match what was in https://github.com/metaplex-foundation/mpl-bubblegum/pull/3

### CODE REVIEW

File: programs/bubblegum/program/src/processor/update_metadata.rs
Comment (danenbm) [inline programs/bubblegum/program/src/processor/update_metadata.rs:1]: Validated these functions all match what was in https://github.com/metaplex-foundation/mpl-bubblegum/pull/3

File: programs/bubblegum/program/src/lib.rs [hunk 1]
```diff
@@ -31,7 +34,10 @@ pub enum InstructionName {
     UnverifyCollection,
     SetAndVerifyCollection,
     MintToCollectionV1,
+    SetDecompressableState,
```
[the remark this answers is not in the document]
Comment (danenbm) [reply programs/bubblegum/program/src/lib.rs]: Removed

File: idls/bubblegum.json [hunk 2]
```diff
@@ -1888,8 +2157,17 @@
           {
             "name": "MintToCollectionV1"
           },
+          {
+            "name": "SetDecompressableState"
+          },
```
[the remark this answers is not in the document]
Comment (danenbm) [reply idls/bubblegum.json]: Fixed by removing from `InstructionName` and regenerating

**Review by blockiosaurus — APPROVED** [review]

**Review by febo — APPROVED** [review]
LGTM

**Pull request merged**